docs/docs/tutorials/first-agent/4-human-in-the-loop.md

```diff
@@ -0,0 +1,331 @@
+# Part 4: Human-in-the-loop
+
+Agents can be unreliable and may need human input to successfully accomplish tasks. Similarly, for some actions, you may want to require human confirmation before sensitive tasks/actions like making a purchase.
+
+LangGraph supports `human-in-the-loop` workflows in a number of ways. In this section, we will use LangGraph's `interrupt_before` functionality to always break the tool node.
+
+We'll be continuing using the final code from [part 3 - adding memory to the chatbot](/first-agent/3-persistent-state.md). If you haven't been following along, make sure you complete the [setup steps](/first-agent/0-setup.md) and copy the code from part 3 before continuing.
+
+## Step 1: Interrupt agent execution
+
+To start, we need to decide _where_ in the graph's execution the agent should wait for human feedback. LangGraph provides two options - _before_ or _after_ a node is run. Let's set up the agent to wait for human feedback before using a tool.
+
+Interrupts don't change the _structure_ of the graph, only how it is _executed_. As a result, we don't need to change the way we build the graph. Instead, we'll specify where we want interrupts to occur when we _compile_ the graph.
+
+Locate the code that builds and compiles the graph in your `chatbot.ts` file. It should look like this:
+
+```ts
+// Define the graph and compile it into a runnable
+export const app = new StateGraph(MessagesAnnotation)
+	.addNode("agent", callModel)
+	.addEdge("__start__", "agent")
+	.addNode("tools", new ToolNode(tools))
```

Comment:
This should be `.addNode("tools", tools)` to match instructions from previous steps.